### DISCUSSION TITLE
Sign up redirect to different page than sign in - callback URL???

### DISCUSSION THREAD

**Discussion by j-lee8** [nextauthjs/next-auth#1856] [Help]
I've augmented the user to take an `isNewUser` field to determine whether or not it's a sign up OR sign in. How do I redirect to a different URL on sign up? The callbackURL is a string field and not a function?

I'd love to know how people have done this. This is what I have which doesn't work:

### login.tsx
```
signIn('google', callbackbackUrl: session?.user?.isNewUser ? "http://localhost:3000/app/gettingstarted" : "http://localhost:3000/app/main" });
```
### _app.tsx

```
function Auth({ children }) {
	const router = useRouter();
	const [session, loading] = useSession();
	const isUser = !!session?.user;

	useEffect(() => {
		if (loading) return; // Do nothing while loading
	  	if (!isUser) router.push('/login'); // If not authenticated, force log in
	}, [isUser, loading]);

	if (isUser) return children;

	return <div>Loading...</div>;
}
```

There is confusion because the callback URL goes to a fixed URL but I need it to be different depending on the flag. Has anybody solved this?

**Comment by balazsorban44** [comment]
When you call `signIn`, the session will obviously not be available at any time. What I would do is something like:

```signIn('google', callbackbackUrl: "http://localhost:3000/app/main" });```

And on that `/app/main` page, have a useEffect hook like this:
```js
const [session, loading] = useSession()
const router = useRouter()

const isLoggedIn = !!session?.user
const isNewUser = session?.user.isNewUser

React.useEffect(() => {
  if(!isLoggedIn) return router.push("/login")
  if(isNewUser) return router.push("/app/gettingstarted")
}, [isLoggedIn, isNewUser, router])
```

**Reply by j-lee8** [reply]
Awesome! Thanks buddy. I'll give this a go :)
